src/OpenTelemetry/Resources/IResourceDetector.cs

```diff
@@ -0,0 +1,30 @@
+// <copyright file="IResourceDetector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Resources
+{
+    /// <summary>
+    /// An interface for Resource detectors.
+    /// </summary>
+    public interface IResourceDetector
```

Comment:
we should keep this internal, until the spec is official about Detectors.

Reply:
This would mean that the `GetResourceFromDetectors` will be internal as well for now.